### DISCUSSION TITLE
whats is NMM and IOS?

### DISCUSSION THREAD

**Discussion by fcakyon** [obss/sahi#625] [Announcements]
In the [paper](https://arxiv.org/abs/2202.06934) we only utilized standard NMS and IOU. The NMM and IOS will be discussed in a follow-up paper.

For now you can consider NMM as non-maximum merging, in other words, it merges the lower scored box instead of suppressing it as in NMS. It is useful when the model confidence threshold is large (as 0.3 to 0.5) and the predictions are confident. It merges the box predictions, corresponding to the same instance, coming from different slices. Note that NMM used in `sahi` is different from [OverFeat](https://arxiv.org/abs/1312.6229) in the sense that, NMM merges the boxes by taking their union while OverFeat merges the boxes by averaging them.

IOS is basically intersection over smaller region, ie intersection area of two box / area of smaller box. It is used as the match criteria while performing NMS and NMM.

All NMM and NMS implementations can be performed via IOU or IOS match criteria by efficient matrix calculations using the `sahi.postprocess` module.

If you have any question regarding the postprocess module of `sahi` feel free to ask them under this post.

**Comment by Grenaf** [comment]
Looking forward to the effect of NMM and IOS about smaller region in your new paper，your first paper is very interesting！

**Comment by Mo4py** [comment]
Do the postprocessing algorithms (NMS, NMM etc.) work based on the bounding box (bbox) or the segmentation mask of each detected mask?

After looking into the code i think it is only based on the bbox coordinates?
Would it not be better to use the mask instead? (So e.g. the IOU is calculated from the intersection of the masks, not the bboxes of each detected instance)


Thanks for every answer and for the great library.

**Reply by fcakyon** [reply]
Current implementations are box-based @Mo4py

**Reply by ClemSc** [reply]
Ah, I just came here for this info @fcakyon. Smaller instance captured by the bounding box of a larger instance of the same class are then going to be merged/suppressed, right ? Any ways to avoid this behavior currently ?

**Comment by bshakya77** [comment]
Hello @fcakyon,

I am looking at the Weighted Boxes Fusion method, which is the advanced post-processing method to mitigate the redundant bounding boxes using weighted average. Is this same as NMM method used in this repository. Why is WBF method not implemented or is there any future enhancement to include WBF as a postprocessing method?

Thank you.

Regards,
Bijay
